blog/_posts/2016-10-10-julia-0.5-highlights.md

```diff
@@ -289,72 +289,71 @@ It's common to write expressions applying multiple vectorized functions to some
 For example, one might write something like:
 
 {% highlight julia %}
-atan2(cos(X), sin(Y))
+max(abs(X), abs(Y))
 {% endhighlight %}
 
-Currently, this allocates two temporary intermediate arrays – one to store `cos(X)` and one for `sin(Y)`.
+This computes the absolute values of each element of `X` and `Y` and takes the larger of the corresponding elements from `abs(X)` and `abs(Y)`.
+In this form, the code allocates two temporary intermediate arrays – one to store `abs(X)` and one for `abs(Y)`.
 If we use vectorized function call syntax, however, these calls are syntactically fused into a *single* call to `broadcast` with an anonymous function.
-In other words, `atan2.(cos.(X), sin.(Y))` becomes
+In other words, `max.(abs.(X), abs.(Y))` becomes
 
 {% highlight julia %}
-broadcast((x, y)->atan2(cos(x), sin(y)), X, Y)
+broadcast((x, y)->max(abs(x), abs(y)), X, Y)
 {% endhighlight %}
 
 This version of the computation avoids allocating any intermediate arrays and performs the entire vectorized computation all at once, directly into the result array.
-We can see this difference in allocation when we benchmark these expressions:
+We can see this difference in memory usage and speed when we benchmark these expressions:
 
 {% highlight julia %}
 julia> using BenchmarkTools
 
 julia> X, Y = rand(1000,1000), rand(1000,1000);
 
-julia> @benchmark atan2(cos(X), sin(Y))
+julia> @benchmark max(abs(X), abs(Y))
 BenchmarkTools.Trial:
   memory estimate:  22.89 mb
-  allocs estimate:  13
-  minimum time:     54.09 ms (1.32% GC)
-  median time:      55.82 ms (4.30% GC)
-  mean time:        55.89 ms (4.25% GC)
-  maximum time:     56.95 ms (4.36% GC)
+  minimum time:     13.95 ms (1.77% GC)
+  median time:      14.17 ms (1.76% GC)
+  mean time:        14.32 ms (1.78% GC)
+  maximum time:     17.15 ms (3.47% GC)
 
-julia> @benchmark atan2.(cos.(X), sin.(Y))
+julia> @benchmark max.(abs.(X), abs.(Y))
 BenchmarkTools.Trial:
   memory estimate:  7.63 mb
-  allocs estimate:  38
-  minimum time:     51.83 ms (0.00% GC)
-  median time:      52.18 ms (0.00% GC)
-  mean time:        52.94 ms (1.48% GC)
-  maximum time:     57.91 ms (4.17% GC)
+  minimum time:     2.84 ms (0.00% GC)
+  median time:      2.98 ms (0.00% GC)
+  mean time:        3.27 ms (18.26% GC)
+  maximum time:     5.96 ms (65.68% GC)
 
-julia> 22.89/7.63
-3.0
+julia> 22.89/7.63, 16.63/3.84
+(3.0,4.330729166666667)
 {% endhighlight %}
 
 I'm using the [`BenchmarkTools`](https://github.com/JuliaCI/BenchmarkTools.jl) package here instead of hand-rolled timing loops. `BenchmarkTools` has been carefully designed to avoid many of the common pitfalls of benchmarking code and to provide sound statistical estimates of how much time and memory your code uses.
```

Comment:
It feels like the writing swapped from inclusive "we" to the separate "I" and "you" here. Previously "We can see this", and now "I'm using" and "As you can see". Maybe stick to "we"?